File: submissions/good_solution.py
import sys


def max_subarray_sum(nums: list[int]) -> int:
    best = current = nums[0]
    for x in nums[1:]:
        current = max(x, current + x)
        best = max(best, current)
    return best


def main() -> None:
    data = sys.stdin.read().split()
    n = int(data[0])
    nums = [int(x) for x in data[1 : 1 + n]]
    print(max_subarray_sum(nums))


if __name__ == "__main__":
    main()

File: submissions/good_solution.js
const data = require("fs").readFileSync(0, "utf8").split(/\s+/).filter(Boolean);
const n = parseInt(data[0], 10);
const nums = data.slice(1, 1 + n).map(Number);

let best = nums[0];
let current = nums[0];
for (let i = 1; i < n; i++) {
  current = Math.max(nums[i], current + nums[i]);
  best = Math.max(best, current);
}
console.log(best);
